### ISSUE TITLE
KeyError: 'created_by_ref'

### ISSUE THREAD

**Issue by tlhampton13** [oasis-open/cti-stix-elevator#162]
Getting a KeyError when converting a package.  I've attached an example program that illustrates the problem.  I'm using stix-elevator v2.0.1.

[key_error.py.zip](https://github.com/oasis-open/cti-stix-elevator/files/3535325/key_error.py.zip)

Here is the traceback:
```
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/helpers/pydev/pydevd.py", line 1758, in <module>
    main()
  File "/Applications/PyCharm.app/Contents/helpers/pydev/pydevd.py", line 1752, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/Applications/PyCharm.app/Contents/helpers/pydev/pydevd.py", line 1147, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/thampton/code/magellan/magellan-stix-converter/key_error.py", line 74, in <module>
    stix2_json = elevator.elevate_string(stix1_xml)
  File "/Users/thampton/code/magellan/magellan-stix-converter/venv/lib/python3.7/site-packages/stix2elevator/__init__.py", line 124, in elevate_string
    json_string = json.dumps(convert_package(stix_package, env),
  File "/Users/thampton/code/magellan/magellan-stix-converter/venv/lib/python3.7/site-packages/stix2elevator/convert_stix.py", line 1848, in convert_package
    finalize_bundle(bundle_instance)
  File "/Users/thampton/code/magellan/magellan-stix-converter/venv/lib/python3.7/site-packages/stix2elevator/convert_stix.py", line 1742, in finalize_bundle
    operation_on_path(bundle_instance, item, "", op=2)
  File "/Users/thampton/code/magellan/magellan-stix-converter/venv/lib/python3.7/site-packages/stix2elevator/utils.py", line 211, in operation_on_path
    operation_on_path(current_obj, path, value, op)
  File "/Users/thampton/code/magellan/magellan-stix-converter/venv/lib/python3.7/site-packages/stix2elevator/utils.py", line 211, in operation_on_path
    operation_on_path(current_obj, path, value, op)
  File "/Users/thampton/code/magellan/magellan-stix-converter/venv/lib/python3.7/site-packages/stix2elevator/utils.py", line 201, in operation_on_path
    current_obj = obj[current]
KeyError: 'created_by_ref'
```

Stepping through the code I notice in stix_converter.finalize_bundle() uses iterpath() to build a list of properties to to remove ```to_remove```.  This list contains duplicates paths when processing the example I've provided.  This causes the ```operation_on_path()``` function to fail when it attempts to remove a property that does not exist.

Not sure why iterpath() is visiting the same path muliplte times ... but at the very least ```operation_on_path()``` should not fail if a property doesn't exist. 

In the screenshot below, from my, i'm inspecting ```to_remove``` at line 1741 just before the loop calling ```operation_on_path()```.  Notice the path ['objects', '[0]', 'created_by_ref'] appears in the list twice.

![Screen Shot 2019-08-23 at 12 30 25 PM](https://user-images.githubusercontent.com/11504292/63607996-daa28000-c5a1-11e9-9c5b-ceee38932d46.png)

**Comment by rpiazza** [comment]
@emmanvg - can you look into this?

**Comment by emmanvg** [comment]
@tlhampton13,  just wanted to let you know that I have been looking into the issue. Your fix in fact solves the problem of the `KeyError` for the empty key/value removal procedure.

As you mentioned, there are other issues with the generated content that also need to be looked into and perhaps @rpiazza knows better about them.

- Vulnerability object duplicated on output
- There is a relationship where `source_ref` and `target_ref` point to the same `identity` ID

Attached JSON output for further review: [key_error_bundle.txt](https://github.com/oasis-open/cti-stix-elevator/files/3560277/key_error_bundle.txt)

**Comment by tlhampton13** [comment]
@emmanvg thanks for the update !

I should mention that after I monkey patched ```iterpath``` and ```operation_on_path``` per my comments above, my conversion was successful.  

The duplication in the ```to_remove``` list did not result in duplication in the converted bundle.  So long as it doesn't fail with the Key Error, I don't know if the  duplication in the ```to_remove``` list matters.

**Comment by emmanvg** [comment]
closed by #164 

The next release of the elevator should have both fixes available. Thanks!

**Comment by rpiazza** [comment]
@tlhampton13 -

Thanks for providing this example.  It contained nested embedded objects, which the elevator couldn't handle.  As @emmanvg said - the fixes will be available in the next release, which will be relatively soon.  I have included the correct STIX 2.0 output.

[issue-162.json.txt](https://github.com/oasis-open/cti-stix-elevator/files/3585031/issue-162.json.txt)

**Comment by tlhampton13** [comment]
@rpiazza @emmanvg -

Thank you for the quick fix !
